### PR TITLE
Add new WIF fields to AWS secret backend

### PR DESCRIPTION
**Pull request by Zlaticanin**
This PR involves adding new fields to the AWS secret backend resource to support configuration of plugin WIF.

### CODE REVIEW

File: vault/resource_aws_secret_backend.go [hunk 1]
```diff
@@ -232,141 +276,156 @@ func awsSecretBackendRead(d *schema.ResourceData, meta interface{}) error {
...
 
 	path, err := util.Remount(d, client, consts.FieldPath, false)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	if d.HasChange("default_lease_ttl_seconds") || d.HasChange("max_lease_ttl_seconds") || d.HasChange("description") {
-		description := d.Get("description").(string)
+	if d.HasChange(consts.FieldDefaultLeaseTTL) || d.HasChange(consts.FieldMaxLeaseTTL) || d.HasChange(consts.FieldDescription) || d.HasChange(consts.FieldIdentityTokenKey) {
```
Comment (fairclothjm) [inline vault/resource_aws_secret_backend.go:347]: nit: I think we could use `d.HasChanges` here

File: website/docs/r/aws_secret_backend.html.md [hunk 2]
```diff
@@ -79,6 +87,14 @@ for credentials issued by this backend.
 
 * `local` - (Optional) Specifies whether the secrets mount will be marked as local. Local mounts are not replicated to performance replicas.
 
+* `identity_token_audience` - (Optional) The audience claim value.
+
+* `identity_token_ttl` - (Optional) The TTL of generated identity tokens in seconds.
+
+* `identity_token_key` - (Optional) The key to use for signing identity tokens.
+
+* `role_arn` - (Optional) Role ARN to assume for plugin identity token federation.
```
Comment (fairclothjm) [inline website/docs/r/aws_secret_backend.html.md:96]:
```suggestion
* `identity_token_audience` - (Optional) The audience claim value. Requires Vault 1.16+.

* `identity_token_ttl` - (Optional) The TTL of generated identity tokens in seconds. Requires Vault 1.16+.

* `identity_token_key` - (Optional) The key to use for signing identity tokens. Requires Vault 1.16+.

* `role_arn` - (Optional) Role ARN to assume for plugin identity token federation. Requires Vault 1.16+.
```

**Review by fairclothjm — COMMENTED** [review]
Could we add a changelong entry? Overall, looks great! Just a few nits :)

File: vault/resource_aws_secret_backend.go (hunk 1 above)
Comment (Zlaticanin) [reply vault/resource_aws_secret_backend.go]: Nice catch! Thank you!

**Review by fairclothjm — APPROVED** [review]
LGTM!

**Review by austingebauer — APPROVED** [review]
LGTM! Going to try this out for my demo :)

**Pull request merged**